Add BUILD_URL and BUILD_URL_SHORT in StepMetadata.Env()

analytically opened the pull request:
This commit:
- Adds a BUILD_URL and BUILD_URL_SHORT env parameters to step metadata
- Adds proper handling for one-off jobs with a different URL pattern
- Adds instance params as query params

Was also requested here: #1588

taylorsilva commented on atc/exec/step_metadata.go:
```diff
@@ -61,6 +61,24 @@ func (metadata StepMetadata) Env() []string {
+			buildURL := buildURLShort
+
+			if metadata.PipelineName != "" && metadata.JobName != "" && metadata.BuildName != "" {
+				buildURL = fmt.Sprintf("%s/teams/%s/pipelines/%s/jobs/%s/builds/%s",
```
This URL would be invalid for instance pipelines. Instance pipelines follow the same URL format, except at the end they have all the instance vars listed as URL params.

For example, the build for one of the checks for this PR is:
```
https://ci.concourse-ci.org/teams/contributor/pipelines/pr/jobs/validate-labels/builds/2?vars.number=9296
```

Note the `?vars.number=9296` at the end. If you strip the instance vars you get a HTTP 404.

taylorsilva reviewed (changes requested):
Need to support instance pipeline URL's.

analytically committed:
Add BUILD_URL in StepMetadata.Env()
This commit:

- Adds a BUILD_URL and BUILD_URL_SHORT env parameters to step metadata
- Adds proper handling for one-off jobs with a different URL pattern
- Adds instance params as query params

Was also requested here: #1588

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>

taylorsilva reviewed (approved):
LGTM - Locally verified as well using the mock resource
